frontend/src/query_builder/ExpressionWidget.jsx

```diff
@@ -0,0 +1,465 @@
+import React, { Component, PropTypes } from 'react';
+
+import _ from 'underscore';
+
+import Icon from "metabase/components/Icon.jsx";
+import Popover from "metabase/components/Popover.jsx";
+
+
+const VALID_OPERATORS = new Set(['+', '-', '*', '/']);
+
+const OPERATOR_SUGGESTIONS = [
+    {name: '+'},
+    {name: '-'},
+    {name: '*'},
+    {name: '/'}
+];
+
+const ERROR_MESSAGE_EMPTY_NAME       = 'what should this expression be called?';
+const ERROR_MESSAGE_EMPTY_EXPRESSION = 'enter an expression';
+
+const KEYCODE_ENTER = 13;
+const KEYCODE_UP    = 38;
+const KEYCODE_DOWN  = 40;
+
+
+function tokenizeExpression(expression, i = 0, level = 0) {
```

Comment:
They're more or less part of the `ExpressionWidget` but they're things that I would make `defn-` / `^:private ^:const` in Clojure. So in this case just making them unexported functions + constants seems to achieve the same effect here